src/app/cube/new-home/help/build-program/build-program.component.ts

```diff
@@ -1,15 +1,192 @@
 import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { LearningObject } from '@entity';
+import { GuidelineService } from 'app/core/guideline.service';
+import { BuildProgramComponentService } from 'app/cube/core/build-program-component.service';
+import { FrameworkDocument } from 'entity/standard-guidelines/Framework';
+import { SearchItemDocument } from 'entity/standard-guidelines/search-index';
 
 @Component({
   selector: 'clark-build-program',
   templateUrl: './build-program.component.html',
-  styleUrls: ['./build-program.component.scss']
+  styleUrls: ['./build-program.component.scss'],
+  animations: [
+    trigger('frameworkView', [
+      transition(':enter', [
+        style({
+          transform: 'translateX(-100%)',
+          opacity: 0
+        }),
+        animate('400ms 0ms ease-out', style({
+          transform: 'translateX(0)',
+          opacity: 1
+        }))
+      ]),
+      transition(':leave', [
+        style({
+          position: 'absolute',
+          willChange: 'contents'
+        }),
+        animate('400ms 0ms ease-out', style({
+          willChange: 'contents',
+          transform: 'translateX(-100%)',
+        }))
+      ])
+    ]),
+    trigger('guidelineView', [
+      transition(':enter', [
+        style({
+          transform: 'translateX(100%)',
+          opacity: 0
+        }),
+        animate('400ms 0ms ease-out', style({
+          transform: 'translateX(0)',
+          opacity: 1
+        }))
+      ]),
+      transition(':leave', [
+        style({
+          position: 'absolute',
+          willChange: 'contents'
+        }),
+        animate('400ms 0ms ease-out', style({
+          willChange: 'contents',
+          transform: 'translateX(100%)',
+          opacity: 0
+        }))
+      ])
+    ])
+  ]
 })
-export class BuildProgramComponent implements OnInit {
+export class BuildProgramComponent implements OnInit{
+  frameworks: FrameworkDocument[];
+  currentFramework: string;
+  currentFrameworkGuidelines: SearchItemDocument[];
 
-  constructor() { }
+   // separated from currentFrameworkGuidelines for searching
+  currentFrameworkGuidelinesFiltered: SearchItemDocument[];
+
+  // separated from currentFrameworkGuidelinesPaginated for maintaining pagination
+  currentFrameworkGuidelinesPaginated: SearchItemDocument[][];
+
+  guidelineSearch = ''; // search bar input
+  currentPage = 0;
+
+  constructor(private buildProgramComponentService: BuildProgramComponentService,
+              private guidelineService: GuidelineService,
+              private router: Router) { }
 
   ngOnInit(): void {
+    this.buildProgramComponentService.currentFrameworkObservable
+    .subscribe(framework => {
+      this.currentFramework = framework;
+      this.guidelineSearch = '';
+      this.currentFrameworkGuidelines = [];
+      this.currentPage = 0;
+    });
+    this.guidelineService.getFrameworks({ limit: 100, page: 1 })
```

Comment:
Why 100?

Reply:
When there's no limit property, the function seems to have a default limit of 10, showing only 10 out of the 15 frameworks we currently have. Other components call this function with a limit of 100 as well, so I just copied that. I would say to keep it at 100 to show all the frameworks and allow for any future additions. What do you think?

Reply:
100 seems a bit too high, how about 30?  Frameworks don't come out often and they get deprecated after a certain amount of years.

Reply:
OK, I'll set it to 30